cosmic/sample/sampler/independent.py

```diff
@@ -226,17 +265,21 @@ def binary_select(self, primary_mass, model='half'):
             primary masses that will be single stars
         """
 
-        if model == 'half':
-            binary_choose = np.random.uniform(0, 1.0, primary_mass.size)
-            binaryIdx, = np.where(binary_choose >= 0.5)
-            singleIdx, = np.where(binary_choose < 0.5)
-
-        elif model == 'vanHaaften':
-            binary_fraction = 1/2.0 + 1/4.0 * np.log10(primary_mass)
-            binary_choose =  np.random.uniform(0, 1.0, primary_mass.size)
+        if type(model) == str:
+            if model == 'vanHaaften':
+                binary_fraction = 1/2.0 + 1/4.0 * np.log10(primary_mass)
+                binary_choose =  np.random.uniform(0, 1.0, primary_mass.size)
 
-            binaryIdx, = np.where(binary_fraction > binary_choose)
-            singleIdx, = np.where(binary_fraction < binary_choose)
+                binaryIdx, = np.where(binary_fraction > binary_choose)
+                singleIdx, = np.where(binary_fraction < binary_choose)
+            else:
+                raise Error('You have supplied a non-supported binary fraction model. Please choose vanHaaften or a float')
+        elif type(model) == float:
```

Comment:
In this line, you should force the binary fraction to be 0 <= bin_frac <= 1, otherwise raise a value error